cinema/views.py

```diff
@@ -87,6 +122,14 @@ class MovieSessionViewSet(viewsets.ModelViewSet):
         )
     )
     serializer_class = MovieSessionSerializer
+    authentication_classes = (TokenAuthentication,)
+    permission_classes = (IsAdminOrIfAuthenticatedReadOnly,)
+
+    def get_permissions(self):
+        if self.action in ("create", "update", "partial_update", "destroy"):
+            return [IsAdminUser()]
+
+        return super().get_permissions()
```

Comment:
why do u need this ?

Reply:
for some reason I thought that it was necessary to override the permissions for specific actions here separately. But I already understood that we already define permission_classes = (IsAdminOrIfAuthenticatedReadOnly,) for this class